console/console-init/ui/src/components/DropdownWithToggle/DropdownWithToggle.tsx

```diff
@@ -74,11 +76,13 @@ export const DropdownWithToggle: React.FC<IDropdownWithToggleProps &
    *  If there is only one option use it as default
    */
   useEffect(() => {
-    if (!value || value.trim() === "") {
-      if (dropdownItems && dropdownItems.length === 1) {
-        const item = dropdownItems[0];
-        if (item && item.value && onSelectItem) {
-          onSelectItem(item.value);
+    if (isRequiredField) {
```

Comment:
I thing we can reduce number of if checks. can try this.
useEffect(() => {
       if(isRequiredField && (!value || value.trim() === "") && dropdownItems?.length === 1){
         const item = dropdownItems[0];
          if (item && item.value && onSelectItem) {
            onSelectItem(item.value);
          }
       }
  }, [dropdownItems]);